test/runnable/testcontracts.d

```diff
@@ -402,6 +402,90 @@ void test7218()
 }
 
 /*******************************************/
+// 7517
+
+void test7517()
+{
+    static string result;
+
+    interface I
+    {
+        static I self;
+
+        void setEnable()
+        in
+        {
+            assert(self is this);
+            result ~= "I.setEnable.in/";
+            assert(!enabled);
+        }
+        out
+        {
+            assert(self is this);
+            result ~= "I.setEnable.out/";
+            assert( enabled);
+        }
+
+        void setDisable()
+        in
+        {
+            assert(self is this);
+            result ~= "I.setDisable.in/";
+            assert( enabled);
+        }
+        out
+        {
+            assert(self is this);
+            result ~= "I.setDisable.out/";
+            assert(!enabled);
+        }
+
+        @property bool enabled() const;
+    }
+
+    class C : I
+    {
+        static C self;
+
+        void setEnable()
+        in {}       // supply in-contract to invoke I.setEnable.in
```

Comment:
Huh, I completely forgot about it :-(. While I think it's more logical, and was a huge source of bugs for me (library methods are using contracts, and my code that was extending them was not, so none of the contracts were executed making me scratch the head when I saw that some arguments inside body are null, despite the `in` contract), I'm not sure if this will actually be accepted, as it *is* a breaking change, and I got busy with another things, so I completely forgot about it. If there's an interest, I can make it before Christmas.